python/mp/effects/sine_wave.py

```diff
@@ -12,8 +13,12 @@ class SineWave(effect.Effect):
     * period: the number of sine wave periods in the unit circle (can be float)
     * direction: the distance (positive or negative float) to advance around the unit circle in one mainloop step
     """
-    def __init__(self, bead_set, color=color.Color(1,1,1), period=1, direction=1):
-        super().__init__("sine_wave", bead_set, color=color)
+
+    # Wish there were a better way than requiring this every time
+    dm = copy.deepcopy(effect.Effect.dm)
+
+    def __init__(self, bead_set, color=color.Color(1,1,1), duration=None, period=1, direction=1):
```

Comment:
It was fine in the `Bounce` effect, but for effects with more args in the `__init__`, duration gets inelegantly shoved in there.